README.md

```diff
@@ -23,9 +23,8 @@ Do not skip this!
 It has important information.
 
 `lsp-mode` automatically requires the `lsp-haskell` package , so you do not need to `require` `lsp-haskell` unless you like being explicit.
-Similarly, `lsp-haskell` automatically requires the `haskell-mode` package, so you do not need to `require` `haskell-mode`.
 
-You will need to set some hooks to ensure that `lsp-mode` is triggered when the `haskell-mode` major mode is entered.
+You will need to set some hooks to ensure that `lsp-mode` is triggered when your Haskell major mode (e.g., `haskell-mode` or `haskell-ts-mode`) is entered.
```

Comment:
Is this actually required? I'm not sure I have this? Is this handled now by the `:major-modes` thing in `lsp-mode` now?

I definitely don't want it to be the case that everyone has to add hooks, but I think maybe they already don't?

Reply:
Hm, not sure if this works. I do see that `haskell-ts-mode` is part of `:major-modes` in `lsp-haskell`, but for me this does not trigger `lsp-mode` in a `haskell-ts-mode` buffer.

I basically have this setup:
```
(use-package haskell-ts-mode
  :mode "\\.hs\\'")

(use-package lsp-haskell
  :after haskell-ts-mode
  :config
  (add-hook 'haskell-ts-mode-hook #'lsp-deferred))
```

If I remove the `add-hook 'haskell-ts-mode` line, `lsp` is not activated in Haskell buffers with `haskell-ts-mode` activated.

My `lsp-mode` setup is quite old (years), so maybe I am missing some new configuration necessary to make the automatic initialization work?

Reply:
No, I think you're right and my emacs setup just does this for me. So I think this is fine.k